oak/server/base_credential_provider.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_SERVER_BASE_CREDENTIAL_PROVIDER_H_
+#define OAK_SERVER_BASE_CREDENTIAL_PROVIDER_H_
+
+#include <memory>
+
+#include "include/grpcpp/grpcpp.h"
+#include "include/grpcpp/security/server_credentials.h"
+
+namespace oak {
+
+// Provides channel credentials for use by gRPC servers.
+class BaseCredentialProvider {
+ public:
+  virtual ~BaseCredentialProvider() {}
+  virtual void Initialize(){};
+  virtual std::shared_ptr<grpc::ServerCredentials> BuildCredentials() const = 0;
```

Comment:
I wondered if a function pointer might be simpler than a class hierarchy, but probably overtaken by the comment about passing credentials rather than credential-builders above.

Reply:
Changed to std::shared_ptr\<grpc::ServerCredentials\>